flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -202,16 +204,27 @@ public YarnClusterDescriptor(
         this.nodeLabel = flinkConfiguration.getString(YarnConfigOptions.NODE_LABEL);
     }
 
-    private Optional<List<File>> decodeFilesToShipToCluster(
+    private Optional<List<Path>> decodeFilesToShipToCluster(
             final Configuration configuration, final ConfigOption<List<String>> configOption) {
         checkNotNull(configuration);
         checkNotNull(configOption);
 
-        final List<File> files =
-                ConfigUtils.decodeListFromConfig(configuration, configOption, File::new);
+        List<Path> files = ConfigUtils.decodeListFromConfig(configuration, configOption, Path::new);
+        files =
+                files.stream()
+                        .map(
+                                path ->
+                                        isLocalPath(path)
+                                                ? new Path(new File(path.toString()).toURI())
+                                                : path)
+                        .collect(Collectors.toList());
         return files.isEmpty() ? Optional.empty() : Optional.of(files);
     }
 
+    private boolean isLocalPath(Path path) {
```

Comment:
The method name and implementation are not consistent.

The method name is `isLocalPath(Path path)`, however the implementation is `isWithoutSchema()`